molviewspec/molviewspec/molstar_widgets.py

```diff
@@ -16,45 +28,48 @@ def molstar_html(state: str | dict | MVSData, data=None):
         # convert state to JSON string
         if hasattr(state, "model_dump_json"):
             # pydantic v1 compatibility
-            state = state.model_dump_json(exclude_none=True)
+            state = state.model_dump_json(exclude_none=True, indent=None)
         else:
-            state = state.json(exclude_none=True)
+            state = state.json(exclude_none=True, indent=None)
+    elif isinstance(state, MVSJ):
+        state = state.dumps(indent=None)
+    elif isinstance(state, MVSX):
+        state = "base64," + base64.b64encode(state.dumps()).decode("utf-8")
+        format = "mvsx"
+    elif isinstance(state, bytes):
+        state = "base64," + base64.b64encode(state).decode("utf-8")
+        format = "mvsx"
     else:
         raise TypeError(f"State should be str, dict, State or States, got: {type(state).__name__}: {state}")
 
-    if data is not None:
+    if not isinstance(state, MVSX) and not isinstance(state, bytes) and data is not None:
```

Comment:
The code about converted `state` to string so this will be always true, no?

Seems suspicious to me (especially, MVSX would be base64-encoded, no?)

Reply:
Right, this requires 
different logic.